src/wasm/extract.hpp

```diff
@@ -48,13 +48,13 @@ class file_output : private boost::noncopyable {
   boost::uint64_t total_written_;
 
   bool write_;
-  Nonzip &zip_;
-	bool zip_open_;
-	uint32_t zipindex_;
+  ZIPstream *zip_ = NULL;
```

Comment:
While it makes sense to use a raw pointer as some sort of a handle to the object, you can't really ever know if the object is still valid. You can either trust that the `ZIPstream` object will outlive the `file_output` or you could use `std::shared_ptr` to make sure `ZIPstream` is alive as long as the last `file_output`. If the object is created via the C function and also freed by the C function, then you'd probably want to create a custom deleter for that. But still, this is just FYI, raw pointer in this case should be fine.

Reply:
Also I suggest using `nullptr` instead of `NULL`, since `NULL` is (or rather was) basically just a `0` define. `nullptr` is an object of `nullptr_t` type, which is specifically designed to be treated as a null pointer for any pointer type. I believe right now `NULL` could also be defined as `nullptr`, but still it's in good taste to write that `nullptr` explicitly and use it consistently.

Reply:
It's also worth noting that you could actually just leave it without an initializer since there isn't really a default constructor for this class, and you already correctly initialize it in the other constructor.

Reply:
Also, in most of the other places the pointer's `*` and reference's `&` stands next to the type rather than the name, so to keep things consistent I'd suggest moving it to the left.
```suggestion
  ZIPstream* zip_ = NULL;
```

Reply:
Ok, let's use nullptr where possible. 
```suggestion
  ZIPstream* zip_ = nullptr;
```